tests/lib/rules/no-return-await.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-return-await");
+const RuleTester = require("../../../lib/testers/rule-tester");
+
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2017 } });
+
+ruleTester.run("no-return-await", rule, {
+
+    valid: [
+        { code: "\nasync function foo() {\n\tawait bar(); return;\n}\n" },
+        { code: "\nasync function foo() {\n\tconst x = await bar(); return x;\n}\n" },
+        { code: "\nasync () => { return bar(); }\n" },
+        { code: "\nasync () => bar()\n" },
+    ],
+
+    invalid: [
+        {
+            code: "\nasync function foo() {\n\treturn await bar();\n}\n",
+            errors: [{
+                message: "`return await` from an `async function` is redundant - just `return` the value directly.",
+                type: "Identifier", // pending https://github.com/eslint/espree/issues/304, should be "Keyword"
+            }],
+        },
+        {
+            code: "\nasync () => { return await bar(); }\n",
+            errors: [{
+                message: "`return await` from an async arrow function is redundant - just `return` the value directly.",
+                type: "Identifier", // pending https://github.com/eslint/espree/issues/304, should be "Keyword"
+            }],
+        },
+        {
+            code: "\nasync () => await bar()\n",
+            errors: [{
+                message: "await in an implicit-return async arrow function is redundant - just return the value directly.",
+                type: "Identifier", // pending https://github.com/eslint/espree/issues/304, should be "Keyword"
```

Comment:
i left in these assertions, with a comment, so that when espree fixes its bug - these tests will fail, and they'll need to be updated to the correct values.